modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js

```diff
@@ -430,22 +430,26 @@ ActionsDropdownRenderer.propTypes = {
 		PropTypes.shape({
 			data: PropTypes.shape({
 				confirmationMessage: PropTypes.string,
-				method: PropTypes.oneOf(['get', 'delete']),
+				method: PropTypes.oneOf(['delete', 'get', 'patch', 'post']),
 				permissionKey: PropTypes.string,
 				successMessage: PropTypes.string,
 			}),
 			href: PropTypes.string,
 			icon: PropTypes.string,
 			label: PropTypes.string.isRequired,
-			method: PropTypes.oneOf(['get', 'delete']),
+			method: PropTypes.oneOf(['delete', 'get', 'patch', 'post']),
 			onClick: PropTypes.string,
 			target: PropTypes.oneOf([
-				'modal',
-				'sidePanel',
-				'link',
 				'async',
 				'headless',
 				'inlineEdit',
+				'link',
+				'modal',
+				'modal-full-screen',
+				'modal-lg',
+				'modal-permissions',
+				'modal-sm',
+				'sidePanel',
```

Comment:
Hey @markocikos, why we are typing with oneOf this string? I think PropTypes.string is enough

Reply:
> why we are typing with oneOf this string? I think PropTypes.string is enough

I strongly dislike that this `target` prop even exists. It's basically a code constant that you pass in and it decides which one of internal implementation is applied. This is bad design for a general purpose component. For one thing, API should be descriptive, for example `inlineEdit` boolean. More importantly, specific implementations should be left to specific usages. We should analyze how these are used. Those who are rarely used, we should move out of the component and into usages, set through `propsTransformer`. The same for those that implement product-specific logic, like `modal-permissions`. Modal size is something that should be defined in the specific usage.

A list of these codes can be helpful to us for this migration, that's the only reason I would keep it. I think that changing this to `PropTypes.string` would also be fine, I don't think it'll be a big problem for us to to find our way through code.